lib/handlers/quote/quote.ts

```diff
@@ -171,13 +171,23 @@ export class QuoteHandler extends APIGLambdaHandler<
       for (const protocolStr of protocolsStr) {
         switch (protocolStr.toLowerCase()) {
           case 'v2':
-            protocols.push(Protocol.V2)
+            if (
+              chainId === ChainId.MAINNET ||
+              !['uniswap-ios', 'uniswap-android'].includes(params.requestQueryParams.source ?? '')
```

Comment:
nit: probably makes sense to separate these strings out into a `MOBILE_REQUEST_SOURCES` array or something